atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/priority/NextTableToSweepProviderTest.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.priority;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+public class NextTableToSweepProviderTest {
+    private NextTableToSweepProvider provider;
+
+    private StreamStoreRemappingSweepPriorityCalculator calculator;
+    private Map<TableReference, Double> priorities;
+    private Optional<TableReference> tableToSweep;
+
+    @Before
+    public void setup() {
+        calculator = mock(StreamStoreRemappingSweepPriorityCalculator.class);
+        priorities = new HashMap<>();
+
+        provider = new NextTableToSweepProvider(calculator);
+    }
+
+    @Test
+    public void calculatorReturnsNoMappings_thenProviderReturnsEmpty() {
+        givenNoPrioritiesReturned();
+
+        whenGettingNextTableToSweep();
+
+        thenProviderReturnsEmpty();
+    }
+
+    @Test
+    public void calculatorReturnsOnlyZeroPriorities_thenProviderReturnsEmpty() {
+        givenPriority(table("table1"), 0.0);
+        givenPriority(table("table2"), 0.0);
+        givenPriority(table("table3"), 0.0);
+
+        whenGettingNextTableToSweep();
+
+        thenProviderReturnsEmpty();
+    }
+
+    @Test
+    public void calculatorReturnsSingleHighestPriority_thenProviderReturnsThatTable() {
+        givenPriority(table("table1"), 0.0);
+        givenPriority(table("table2"), 20.0);
+        givenPriority(table("table3"), 30.0);
+        givenPriority(table("table4"), 10.0);
+        givenPriority(table("table5"), 0.0);
+
+        whenGettingNextTableToSweep();
+
+        thenTableChosenIs(table("table3"));
+    }
+
+    @Test
+    public void calculatorReturnsManyTablesWithHighestPriority_thenProviderReturnsOneOfThose() {
+        givenPriority(table("table1"), 0.0);
+        givenPriority(table("table2"), 30.0);
+        givenPriority(table("table3"), 30.0);
+        givenPriority(table("table4"), 30.0);
+        givenPriority(table("table5"), 10.0);
+
+        whenGettingNextTableToSweep();
+
+        Assert.assertTrue(tableToSweep.isPresent());
+        Assert.assertThat(tableToSweep.get(), CoreMatchers.anyOf(
+                CoreMatchers.is(table("table2")), CoreMatchers.is(table("table3")), CoreMatchers.is(table("table4"))));
+    }
```

Comment:
I got used to using CoreMatchers because I used to import * and this has less chance on conflicts.  When I found Matchers was allowed I tidied some things up but not this class.  Doing it now